pkg/executor/core/executor.go

```diff
@@ -82,39 +84,47 @@ func (executor Executor) verifySubjectInternal(ctx context.Context, verifyParame
 	subjectReference.Digest = desc.Digest
 
 	var verifierReports []interface{}
+	eg := errgroup.Group{}
+	var mu sync.Mutex
 
 	for _, referrerStore := range executor.ReferrerStores {
-		var continuationToken string
-		for {
-			referrersResult, err := referrerStore.ListReferrers(ctx, subjectReference, verifyParameters.ReferenceTypes, continuationToken, desc)
-			if err != nil {
-				return types.VerifyResult{}, err
-			}
-			continuationToken = referrersResult.NextToken
-
-			for _, reference := range referrersResult.Referrers {
-
-				if executor.PolicyEnforcer.VerifyNeeded(ctx, subjectReference, reference) {
-					verifyResult := executor.verifyReference(ctx, subjectReference, desc, reference, referrerStore)
-					verifierReports = append(verifierReports, verifyResult.VerifierReports...)
-
-					if !verifyResult.IsSuccess {
-						result := types.VerifyResult{IsSuccess: false, VerifierReports: verifierReports}
-						if !executor.PolicyEnforcer.ContinueVerifyOnFailure(ctx, subjectReference, reference, result) &&
```

Comment:
we don't since we are removing support for short circuiting early during the execution process. It doesn't really make sense anymore since all of these verifications will be happening concurrently. The only use case where its helpful is if one verification for an artifact is taking a lot longer than another. Then if we need to short circuit based on previous result, we could early terminate that long running verification. This would require adding some channels so the routines can communicate. Right now there really isn't a need for this with our verifiers. We discussed earlier and decided we would add such a feature later if needed.

Reply:
I've left the implementation is the ORAS store but should we just remove it?

Reply:
Going to leave this in for now in ORAS store. we can follow up and remove it if we decide to commit to this approach.